tests/integration/test_alert_propagation.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import json
+import logging
+from types import SimpleNamespace
+
+import pytest
+import pytimeparse
+import yaml
+from helpers import get_address, oci_image
+from pytest_operator.plugin import OpsTest
+from workload import Mimir
+
+logger = logging.getLogger(__name__)
+
+mimir = SimpleNamespace(
+    name="mimir", resources={"mimir-image": oci_image("./metadata.yaml", "mimir-image")}
+)
+tester = SimpleNamespace(charm="avalanche-k8s", name="avalanche")
+
+
+@pytest.mark.abort_on_fail
+async def test_deploy_and_relate_charms(ops_test: OpsTest, mimir_charm):
+    """Test that Mimir can be related with Prometheus over prometheus_scrape."""
+    # Build charm from local source folder
+    # mimir_charm = await ops_test.build_charm(".")
+    await asyncio.gather(
+        ops_test.model.deploy(
+            mimir_charm,
+            resources=mimir.resources,
+            application_name=mimir.name,
+            trust=True,
+        ),
+        ops_test.model.deploy(
+            tester.charm,
+            application_name=tester.name,
+            channel="edge",
+        ),
+    )
+
+    await ops_test.model.add_relation(f"{mimir.name}:receive-remote-write", tester.name)
+    await ops_test.model.wait_for_idle(status="active")
+
+
+@pytest.mark.abort_on_fail
+async def test_rules_are_loaded(ops_test):
+    address = await get_address(ops_test, mimir.name, 0)
+    client = Mimir(host=address)
+
+    rules = json.loads(await client.api_request("/prometheus/api/v1/rules"))
+    # Response looks like this:
+    # {"status":"success","data":{"groups":[{"name":"test-alert-propagation-...
+    assert len(rules["data"]["groups"]) > 0
```

Comment:
I wonder if we actually need this, or whether the check for Mimir should be via `/ruler/rule_groups` and YAML.

The prometheus-compatible endpoint is convenient for users who want to use Mimir like prom, but for our integration testing, does it add any value over the ruler groups? We need to check those regardless, and it may be the same information.

Or, since we're not multi-tenant so it's anonymous, we could:

```python
assert len(groups["anonymous"]) > 0
assert rules["data"]["groups"] == groups["anonymous"]`
```

Reply:
The check is very quick. It also doesn't harm: straightforward and documented. I'm fine either way (keeping or removing).

I wouldn't `assert rules["data"]["groups"] == groups["anonymous"]` because of the nested dict/list stuff, and because this would be a mimir workload test,  not a mimir charm test.

Reply:
Right, but the workload is the one with both endpoints to compare for equality.

Reply:
(Itest just failed for this assertion after the last commit. Looking into it.)